mitosheet/mitosheet/code_chunks/step_performers/import_steps/excel_range_import_code_chunk.py

```diff
@@ -5,22 +5,105 @@
+]
+
 EXCEL_RANGE_END_CONDITION_FIRST_EMPTY_VALUE = 'first empty cell'
 EXCEL_RANGE_END_CONDITION_BOTTOM_LEFT_CORNER_VALUE = 'bottom left corner value'
+EXCEL_RANGE_END_CONDITION_BOTTOM_LEFT_CORNER_VALUE_STARTS_WITH = 'bottom left corner value starts with'
+EXCEL_RANGE_END_CONDITION_BOTTOM_LEFT_CORNER_VALUE_CONTAINS = 'bottom left corner value contains'
+EXCEL_RANGE_END_CONDTIONS = [
+    EXCEL_RANGE_END_CONDITION_FIRST_EMPTY_VALUE,
+    EXCEL_RANGE_END_CONDITION_BOTTOM_LEFT_CORNER_VALUE,
+    EXCEL_RANGE_END_CONDITION_BOTTOM_LEFT_CORNER_VALUE_STARTS_WITH,
+    EXCEL_RANGE_END_CONDITION_BOTTOM_LEFT_CORNER_VALUE_CONTAINS
+]
+
+EXCEL_RANGE_COLUMN_END_CONDITION_FIRST_EMPTY_CELL = 'first empty cell'
 EXCEL_RANGE_COLUMN_END_CONDITION_NUM_COLUMNS = 'num columns'
+EXCEL_RANGE_COLUMN_END_CONDITIONS = [
+    EXCEL_RANGE_COLUMN_END_CONDITION_FIRST_EMPTY_CELL,
+    EXCEL_RANGE_COLUMN_END_CONDITION_NUM_COLUMNS
+]
+
+def get_table_range_params(file_path: str, sheet_name: str, start_condition: Any, end_condition: Any, column_end_condition: Any) -> Dict[str, Any]:
+
+    if start_condition['type'] not in EXCEL_RANGE_START_CONDITIONS:
+        raise ValueError(f'Invalid start condition type: {start_condition["type"]}')
+    if end_condition['type'] not in EXCEL_RANGE_END_CONDTIONS:
+        raise ValueError(f'Invalid end condition type: {end_condition["type"]}')
+    if column_end_condition['type'] not in EXCEL_RANGE_COLUMN_END_CONDITIONS:
+        raise ValueError(f'Invalid column end condition type: {column_end_condition["type"]}')
+
+    if start_condition['type'] == EXCEL_RANGE_START_CONDITION_UPPER_LEFT_VALUE:
+        upper_left_value = start_condition['value']
+    else:
+        upper_left_value = None
+    
+    if start_condition['type'] == EXCEL_RANGE_START_CONDITION_UPPER_LEFT_VALUE_STARTS_WITH:
+        upper_left_value_starts_with = start_condition['value']
+    else:
+        upper_left_value_starts_with = None
+
+    if start_condition['type'] == EXCEL_RANGE_START_CONDITION_UPPER_LEFT_VALUE_CONTAINS:
+        upper_left_value_contains = start_condition['value']
+    else:
+        upper_left_value_contains = None
+    
+    if end_condition['type'] == EXCEL_RANGE_END_CONDITION_BOTTOM_LEFT_CORNER_VALUE:
+        bottom_left_value = end_condition['value']
+    else:
+        bottom_left_value = None
+
+    if end_condition['type'] == EXCEL_RANGE_END_CONDITION_BOTTOM_LEFT_CORNER_VALUE_STARTS_WITH:
+        bottom_left_value_starts_with = end_condition['value']
+    else:
+        bottom_left_value_starts_with = None
+
+    if end_condition['type'] == EXCEL_RANGE_END_CONDITION_BOTTOM_LEFT_CORNER_VALUE_CONTAINS:
+        bottom_left_value_contains = end_condition['value']
+    else:
+        bottom_left_value_contains = None
+
+    if column_end_condition['type'] == EXCEL_RANGE_COLUMN_END_CONDITION_NUM_COLUMNS:
+        num_columns = column_end_condition['value']
+    else:
+        num_columns = None
+
+    all_params = {
```

Comment:
We'd just need to update the type in this case. Since the type is used in one place only, we really won't benefit from this change - we'd have to remember to type it correctly exactly once in either case.